core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -103,34 +122,45 @@ public void close() throws IOException {
      */
     public void abort() throws IOException {
         close();
-        tmpFile.delete();
+        Files.deleteIfExists(tmpFile);
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
-            try {
-                Util.deleteFile(destFile);
-            } catch (IOException x) {
-                tmpFile.delete();
-                throw x;
-            }
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpFile, destFile, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
Where do you see that @jtnord? Not [here](http://docs.oracle.com/javase/8/docs/api/java/nio/file/StandardCopyOption.html#ATOMIC_MOVE).